e3sm_diags/plot/cartopy/zonal_mean_2d_plot.py

```diff
@@ -1,293 +1,187 @@
-from __future__ import print_function
-
-import os
+from typing import List, Optional, Tuple
 
 import matplotlib
 import numpy as np
-import numpy.ma as ma
-from cartopy.mpl.ticker import LatitudeFormatter
+import xarray as xr
+import xcdat as xc
 
-from e3sm_diags.driver.utils.general import get_output_dir
+from e3sm_diags.driver.utils.type_annotations import MetricsDict
 from e3sm_diags.logger import custom_logger
-from e3sm_diags.plot import get_colormap
+from e3sm_diags.parameter.core_parameter import CoreParameter
+from e3sm_diags.parameter.zonal_mean_2d_parameter import DEFAULT_PLEVS
+from e3sm_diags.plot.utils import (
+    DEFAULT_PANEL_CFG,
```

Comment:
For most of the small utility functions, I actually have a different view. I think the original plotting code are somehow more readable and perhaps easier to change...

Reply:
There was a lot of copy and pasting of the same code in different plotters. I extracted those into utility functions that can be re-used if needed by a set's plotter. Otherwise a set's plotter can have its own unique, custom code if it is really different.

Reply:
I've been using VS Code's diff viewer, which makes it easy to compare files to find similar/same code